### PR TITLE
FEATURE: add period filter in top topics route for tags.

### PR DESCRIPTION
**Pull request by vinothkannans**
And also move all the "top topics by period" routes to query string param.

`/top/monthly` => `/top?period=monthly`
`/c/:slug/:id/l/top/monthly` => `/c/:slug/:id/l/top?period=monthly`
`/tag/:slug/l/top/daily` => `/tag/:slug/l/top?period=daily` _(new)_

### CODE REVIEW

**Comment by vinothkannans** [comment]
Few JS tests are failing. Checking it...

**Comment by SamSaffron** [comment]
@vinothkannans is this ready for review, looks like some tests are still failing?

**Comment by vinothkannans** [comment]
@SamSaffron yes, few other tests are failing. I will fix it soon.

**Comment by vinothkannans** [comment]
The "plugins backend" tests are failing. It returns an error in "discourse-solved" plugin if I run "`plugin:spec`" rake tasks. But it's passing the test if I run the task with plugin name "`plugin:spec["discourse-solved"]`". And the fails totally unrelated to the change. I'm not sure about how to locate the error. Still looking 🔎 👀

**Comment by vinothkannans** [comment]
Looks like the fails are unrelated to the PR. It's ready to review.

File: app/assets/javascripts/discourse/app/routes/app-route-map.js [hunk 1]
```diff
@@ -27,27 +27,7 @@ export default function () {
   });
 
   this.route("discovery", { path: "/", resetNamespace: true }, function () {
-    // top
-    this.route("top");
```
Comment (eviltrout) [inline app/assets/javascripts/discourse/app/routes/app-route-map.js:31]: If we remove all these routes, won't a link handled by the javascript app break? I understand server side routes have a redirect but I'm wondering about a link in a post?

Comment (vinothkannans) [reply app/assets/javascripts/discourse/app/routes/app-route-map.js:31]: Yes, we have to redirect those routes in client-side too 👍 
`/top/monthly` => `/top?period=monthly`

**Review by eviltrout — APPROVED** [review]

**Pull request merged**